p2p/src/connection/outgoing_effectful/p2p_connection_outgoing_effectful_effects.rs

```diff
@@ -0,0 +1,67 @@
+use redux::ActionMeta;
+
+use crate::{
+    connection::{
+        outgoing::{
+            P2pConnectionOutgoingAction, P2pConnectionOutgoingInitOpts, P2pConnectionOutgoingState,
+        },
+        P2pConnectionService, P2pConnectionState,
+    },
+    webrtc, P2pPeerStatus,
+};
+
+use super::P2pConnectionOutgoingEffectfulAction;
+
+impl P2pConnectionOutgoingEffectfulAction {
+    pub fn effects<Store, S>(self, _meta: &ActionMeta, store: &mut Store)
+    where
+        Store: crate::P2pStore<S>,
+        Store::Service: P2pConnectionService,
+    {
+        match self {
+            P2pConnectionOutgoingEffectfulAction::RandomInit => {
+                let peers = store.state().disconnected_peers().collect::<Vec<_>>();
+                let picked_peer = store.service().random_pick(&peers);
```

Comment:
I was wondering what happens if peers is empty and just noted that `random_pick` unwraps. We probably need a better way to handle this

Reply:
List of peers not being empty is currently handled by enabling conditions, I could make `random_pick` return `Option` to remove unwrap and add `bug_condition` if it is empty.